### ISSUE TITLE
Rounding option for rectangle corners

### ISSUE THREAD

**Issue by trych** [basiljs/basil.js#214]
This came up in class as well today.

In processing and in p5.js the `rect()` commands already have the option to create round corners. basil.js should also include this, I think.

Could be used for stuff like [this](https://i1.wp.com/daks2k3a4ib2z.cloudfront.net/55459cebb2d38dc972871f70/554ad41cabf193924ca3e03d_01-Sufjan-Stevens-Carrie-%26-Lowell-20150605.jpg?resize=755%2C1111&ssl=1).

**Comment by fabianmoronzirfas** [comment]
```
Syntax	
b.rect(a, b, c, d)
b.rect(a, b, c, d, r)
b.rect(a, b, c, d, tl, tr, br, bl)
Parameters	
a	float: x-coordinate of the rectangle by default
b	float: y-coordinate of the rectangle by default
c	float: width of the rectangle by default
d	float: height of the rectangle by default
r	float: radii for all four corners
tl	float: radius for top-left corner
tr	float: radius for top-right corner
br	float: radius for bottom-right corner
bl	float: radius for bottom-left corner
```
[source](https://processing.org/reference/rect_.html)

**Comment by trych** [comment]
Implemented via #217 and successfully used in my class right now. 😜

Closing this.
